src/V3PerPackage/PerWorkerProcessor.cs

```diff
@@ -0,0 +1,177 @@
+                    }
+                    catch (StorageException ex) when (ex.RequestInformation?.HttpStatusCode == (int)HttpStatusCode.NotFound)
+                    {
+                        // Ignore this error. The message has already been removed.
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private static int GetDequeueCount(StorageQueueMessage<PackageMessage> lastMessage)
+        {
+            var storageQueueMessage = lastMessage
+                .GetType()
+                .GetProperty("Message", BindingFlags.Instance | BindingFlags.NonPublic)?
+                .GetValue(lastMessage);
+
+            var cloudQueueMessage = storageQueueMessage?
+                .GetType()
+                .GetProperty("Message", BindingFlags.Instance | BindingFlags.NonPublic)
+                .GetValue(storageQueueMessage) as CloudQueueMessage;
+
+            return cloudQueueMessage?.DequeueCount ?? 1;
+        }
+
+        private async Task<bool> ProcessPackagesAsync(PerBatchContext batchContext, List<PerPackageContext> packageContexts)
+        {
+            var packagesForLogging = packageContexts
+                .Select(x => new { Id = x.PackageId, Version = x.PackageVersion })
+                .ToList();
+
+            bool complete;
+            try
+            {
+
+                complete = await _perBatchProcessor.ProcessAsync(batchContext, packageContexts);
+
+                if (complete)
+                {
+                    foreach (var package in packageContexts)
+                    {
+                        _logger.LogInformation(
+                            "Package {PackageId}/{PackageVersion} was completed by worker {WorkerName}, process {ProcessName}.",
+                            package.PackageId,
+                            package.PackageVersion,
+                            batchContext.Worker.Name,
+                            batchContext.Process.Name);
+                    }
+                }
+                else
+                {
+                    _logger.LogWarning(
+                        "Packages {Packages} were not completed by worker {WorkerName}, process {ProcessName}.",
+                        packagesForLogging,
+                        batchContext.Worker.Name,
+                        batchContext.Process.Name);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(
+                    0,
+                    ex,
+                    "An exception was thrown while processing packages {Packages}, worker {WorkerName}, process {ProcessName}.",
+                    packagesForLogging,
+                    batchContext.Worker.Name,
+                    batchContext.Process.Name);
+
+                complete = false;
+            }
+
+            try
+            {
+
+                await _perBatchProcessor.CleanUpAsync(batchContext, packageContexts);
```

Comment:
>await [](start = 16, length = 5)

There's an extra line up above this line

Reply:
Fixed.